app/models/role.rb

```diff
@@ -0,0 +1,10 @@
+class Role < ActiveRecord::Base
+  has_and_belongs_to_many :users, :join_table => :users_roles
+  belongs_to :resource, :polymorphic => true
```

Comment:
=> - this is old syntax for ruby 1.8.7 and below. Use it like polymorphic: true. Change it for all values here